sympy/integrals/integrals.py

```diff
@@ -396,6 +397,8 @@ def doit(self, **hints):
             return S.Zero
 
         # now compute and check the function
+        if isinstance(self.function, Sum):
+            return Sum(Integral(self.function.function, self.variables[0]).doit(), self.function.limits)
```

Comment:
That's true, but I don't think we have any good way to detect such cases. The same is true of derivatives of integrals, but we commute those as well. It's probably more useful to commute the two and get a useful result in the cases where they are equal than to get no result ever.